nsv13/code/modules/overmap/fighters/fighters.dm

```diff
@@ -0,0 +1,355 @@
+			W.forceMove(src)
+			update_stats()
+		else if(istype(W, /obj/item/twohanded/required/fighter_component/engine))
+			var/e = 0
+			for(var/obj/item/twohanded/required/fighter_component/engine/en in contents)
+				e++
+			if(e < 2)
+				to_chat(user, "<span class='notice'>You start installing [W] in [src]...</span>")
+				if(!do_after(user, 2 SECONDS, target=src))
+					return
+				to_chat(user, "<span class='notice'>You install [W] in [src].</span>")
+				W.forceMove(src)
+				update_stats()
+
+/obj/structure/overmap/fighter/attack_hand(mob/user)
+	.=..()
+	if(maint_state == MS_OPEN)
+		user.set_machine(src)
+		var/dat
+		dat += "<h2> Overview: </h2>"
+//hp, fuel etc go here
+		dat += "<p>Structural Integrity:[obj_integrity/max_integrity*(100)]%</p>"
+		dat += "<p>Fuel Capacity:%</p>"
+		dat += "<h2> Payload: </h2>"
+//Guns, ammo and torpedos
+		var/atom/movable/pw = get_part(/obj/item/twohanded/required/fighter_component/primary_cannon)
+		if(pw == null)
+			dat += "<p><b>PRIMARY WEAPON NOT INSTALLED</font></p>"
+		else
+			dat += "<a href='?src=[REF(src)]:primary_weapon=1'>[pw?.name]</a><br>"
+		dat += "<p>Ammo Capacity:</p>"
+		var/t = 0
+		for(var/obj/structure/munition/mu in contents)
+			dat += "<a href='?src=[REF(src)];torpedo=1'>[mu?.name]</a><br>"
+			t++
+		switch(t)
+			if(1)
+				dat += "<p><b>ONE TORPEDO PYLON EMPTY</font></p>"
+			if(0)
+				dat += "<p><b>TWO TORPEDO PYLONS EMPTY</font></p>"
+		dat += "<h2> Components: </h2>"
+		var/atom/movable/ap = get_part(/obj/item/twohanded/required/fighter_component/armour_plating)
+		if(ap == null)
+			dat += "<p><b>ARMOUR PLATING NOT INSTALLED</font></p>"
+		else
+			dat += "<a href='?src=[REF(src)];armour_plating=1'>[ap?.name]</a><br>"
+		var/atom/movable/ft = get_part(/obj/item/twohanded/required/fighter_component/fuel_tank)
+		if(ft == null)
+			dat += "<p><b>FUEL TANK NOT INSTALLED</font></p>"
+		else
+			dat += "<a href='?src=[REF(src)];fuel_tank=1'>[ft?.name]</a><br>"
+		var/atom/movable/fl = get_part(/obj/item/fighter_component/fuel_lines)
+		if(fl == null)
+			dat += "<p><b>FUEL LINES NOT INSTALLED</font></p>"
+		else
+			dat += "<a href='?src=[REF(src)];fuel_lines=1'>[fl?.name]</a><br>"
+		var/atom/movable/ts = get_part(/obj/item/fighter_component/targeting_sensor)
+		if(ts == null)
+			dat += "<p><b>TARGETING SENSOR NOT INSTALLED</font></p>"
+		else
+			dat += "<a href='?src=[REF(src)];targeting_sensor=1'>[ts?.name]</a><br>"
+		var/e = 0
+		for(var/obj/item/twohanded/required/fighter_component/engine/en in contents)
+			dat += "<a href='?src=[REF(src)];engine=1'>[en?.name]</a><br>"
+			e++
+		switch(e)
+			if(1)
+				dat += "<p><b>ONE ENGINE NOT INSTALLED</font></p>"
+			if(0)
+				dat += "<p><b>TWO ENGINES NOT INSTALLED</font></p>"
+		var/datum/browser/popup = new(user, "fighter", name, 400, 600)
+		popup.set_content(dat)
+		popup.open()
+		return TRUE
+	else if(maint_state == !MS_OPEN && piloted == FALSE) //temp behaviour - button will break control of fighter

```

Comment:
Consider this, unless this was deliberate?
```suggestion
	else if(maint_state != MS_OPEN && !pilot) //temp behaviour - button will break control of fighter
```

Reply:
I believe it was because I had it all as a switch for maint_state and discovered issues when trying to add another variable.